js/index.js

```diff
@@ -215,107 +216,161 @@ webix.ready(function(){
-              users_list.sort("#name#","desc","string");
-            }
+        view:"text",
+        id:"list_input",
+        on:{
+          onTimedKeyPress:function(){
+            var value = this.getValue().toLowerCase();
+            users_list.filter(function(obj){
+                return obj.name.toLowerCase().indexOf(value) !== -1;
+            });
           }
-        ]
+        }
       },
-      {
-        view:"list",
-        css:"user_list",
-        id:"list",
-        height:300,
-        scrollY:true,
-        scrollX:false,
-        select:true,
-        template:"#name# <span class='remove_list_item_btn webix_icon mdi mdi-close'></span>",
-        onClick:{
-          remove_list_item_btn:function(e, id){
-            this.remove(id);
-            return false;
-          }
-        },
-        url:users_data_link
+      {view:"button", value:"Sort asc", css:"webix_primary", width:120,
+        click:function(){
+          webix.message({text:"Sort asc", expire:350});
+          users_list.sort("#name#","asc","string");
+        }
+      },
+      {view:"button", value:"Sort desc", css:"webix_primary", width:120,
+        click:function(){
+          webix.message({text:"Sort desc", expire:350});
+          users_list.sort("#name#","desc","string");
+        }
+      },
+      {view:"button", value:"Add new", css:"webix_primary", width:120,
+        click:function(){
+          webix.message({text:"New user was added", expire:350});
+
+          users_list.add({name:"John Smit", age:getRandomInt(18,60), country:countries_data.data.pull[getRandomInt(1,8)]["value"]});
+
+          //custom version how to select and show new added element
+          scrollToLastAddedElement(users_list,true,true);
+        }
       }
     ]
   };
   var chart = {
     view:"chart",
+    id:"users_chart",
     type:"bar",
     value:"#age#",
-    label:"#name#",
     minHeight:300,
-    border:true,
     barWidth:40,
     radius:1,
     xAxis:{
-        template:"#age#",
-        title:"Age"
+      template:"#country#",
+      title:"Country"
     },
-    gradient:"falling",
-    url:users_data_link
+    yAxis:{
+      start:0,
+      end:10,
+      step:2
+    }
   };
   var treetable = {
     view:"treetable",
+    //id:"tree",
+    scheme:{
+      $init:function(obj){
+        obj.open = true;
+      }
+    },
     scrollY:true,
     scrollX:false,
     select:true,
     columns:[
       { id:"id", header:"", css:"treetable_id", width:50},
-      { id:"title",	header:"Title", css:"treetable_title", minWidth:300, template:"{common.treetable()} #title#"},
-      { id:"price",	header:"Price", minWidth:300, css:"treetable_price", fillspace:true}
+      { id:"title",	header:"Title", editor:"text", css:"treetable_title", minWidth:300, liveEdit:true, template:"{common.treetable()} #title#"},
+      { id:"price",	header:"Price", editor:"text", minWidth:300, css:"treetable_price", liveEdit:true, fillspace:true}
     ],
+    on:{
```

Comment:
first-level items are also edited and validation fails (expected)
to cancel editing to particular items, use the onBeforeEditStart event. inside it you can get the data item and check its `$count` property. this helper property stores the number of kids, and it is 0 for branches

Reply:
Using this event, I can just cancel editing data of 1 level (whole row)! How to cancel editing data just in 1 cell (empty cell) of this row?

Reply:
id is in object that stores cell coordinates `{row: 2, column: "title"}`, so you can check for column id as well

Reply:
Problem is solved by next code:
onBeforeEditStart:function(id){
        if ( this.getItem(id).$level === 1 && id.column == "price"){
          return false;
        }
      },
      onValidationError:function(id, obj, details){
        if(details.price === true){
          return false;
        }
      }